src/objective-c/GRPCClient/GRPCCall.m

```diff
@@ -139,8 +146,10 @@ - (void)finishWithError:(NSError *)errorOrNil {
   _self = nil;
 
   // If there were still request messages coming, stop them.
-  _requestWriter.state = GRXWriterStateFinished;
-  _requestWriter = nil;
+  @synchronized(_requestWriter) {
```

Comment:
I'm surprised Xcode's 'Analyze' command didn't flag this: _requestWriter is sometimes nil (in fact it is set to nil inside the block) and you should never @synchronize(nil)

did you mean:
@synchronized(self) { ... } ?

since you need to synchronize on an object with a lifetime longer than the @synchronized block